tslearn/bases.py

```diff
@@ -0,0 +1,236 @@
+        sklearn.utils.validation.check_is_fitted
+
+        Returns
+        -------
+        bool
+        """
+        pass
+
+    @abstractmethod
+    def _get_model_params(self):
+        """Get model parameters that are sufficient to recapitulate it."""
+        pass
+
+    def _to_dict(self, output=None):
+        """
+        Get model hyper-parameters and model-parameters
+        as a dict that can be saved to disk.
+
+        Returns
+        -------
+        params : dict
+            dict with relevant attributes sufficient to describe the model.
+        """
+
+        if not self._is_fitted():
+            raise ValueError("Model must be fit before it can be packaged")
+
+        d = {'hyper_params': self.get_params(),
+             'model_params': self._get_model_params()}
+
+        # This is just for json support to convert numpy arrays to lists
+        if output == 'json':
+            d['model_params'] = BaseModelPackage._listify(d['model_params'])
+
+        elif output == 'hdf5':
+            d['hyper_params'] = BaseModelPackage._none_to_str(d['hyper_params'])
+
+        return d
+
+    @staticmethod
+    def _none_to_str(mp):
+        """Use str to store Nones. Used for HDF5"""
+        for k in mp.keys():
+            if mp[k] is None:
+                mp[k] = 'None'
+
+        return mp
+
+    @staticmethod
+    def _listify(model_params):
+        """
+        Convert all numpy arrays in model-parameters to lists.
+        Used for json support
+        """
+        for k in model_params.keys():
+            param = model_params[k]
+
+            if isinstance(param, np.ndarray):
+                model_params[k] = param.tolist()  # for json support
+            else:
+                model_params[k] = param
+        return model_params
+
+    @staticmethod
+    def _organize_model(cls, model):
+        """
+        Instantiate the model with all hyper-parameters,
+        set all model parameters and then return the model.
+
+        Do not use directly. Use the designated classmethod to load a model.
+
+        Parameters
+        ----------
+        cls : instance of model that inherits from `BaseModelPackage`
+            a model instance
+
+        model : dict
+            Model dict containing hyper-parameters and model-parameters
+
+        Returns
+        -------
+        model: instance of model that inherits from `BaseModelPackage`
+            instance of the model class with hyper-parameters and
+            model parameters set from the passed model dict
+        """
+
+        model_params = model.pop('model_params')
+        hyper_params = model.pop('hyper_params')  # hyper-params
+
+        # instantiate with hyper-parameters
+        inst = cls(**hyper_params)
+
+        # set all model params
+        for p in model_params.keys():
+            setattr(inst, p, model_params[p])
+
+        return inst
+
+    def to_hdf5(self, path):
+        """
+        Save this model as an HDF5 file.
```

Comment:
Let's use a uniform phrasing for all `to_*` and `from_*` methods. I would suggest:
* Save model to a XXX file
* Load model from a XXX file